### PR TITLE
Implementing MapsService.js and commenting replaced functions in PlaceCtrl

### PR DESCRIPTION
**Pull request by jfking**

### CODE REVIEW

**Comment by jfking** [comment]
Hm it seems like it's not suggested to use $scope in your service but I can't find anything that says you can't:
http://stackoverflow.com/questions/22898927/injecting-scope-into-an-angular-service-function
Doing it the way this guy suggests would hurt the service's ability to avoid code reuse, though

**Comment by jfking** [comment]
Also, do I have to declare the singleton MapsService, or do I just inject it like the other dependencies?

File: parse/public/js/LocationProvider.js [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
   /**
    * Instantiates a LocationProvider.
    * Parameters:
    *   map{google.maps.Map}: The map to plot user location on
    */
-  function LocationProvider(map) {
+  function LocationProvider() {
     console.log("Location Provider instantiated");
-    this._map = map;
+    this._map = MapsService.initMap();
```
Comment (WaseemTheDream) [inline parse/public/js/LocationProvider.js]: `initMap();` doesn't return anything so you're getting null here

File: parse/public/js/MapsService.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+angular.module('atlasApp').service('MapsService', function(
+	$scope, 
+	$http
+) {
+    return {
+
+    	var self = this;
```
Comment (WaseemTheDream) [inline parse/public/js/MapsService.js]: You cannot have declarative statements inside a `return { ... }`.

File: parse/public/js/MapsService.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        /**
+	    * Initializes the Google Maps canvas
+	    */
+	    initMap: function () {
```
Comment (WaseemTheDream) [inline parse/public/js/MapsService.js]: `initMap()` should only be called once in the entirety of the app. 
Whenever `BusInfoProvider` or `UserLocationProvider` need the map object. It should be retrieved from a getMap() function that returns the `google.maps.Map` object.

**Comment by WaseemTheDream** [comment]
I left some comments above. Overall, I recommend using the structure I wrote earlier today in this Gist: https://gist.github.com/WaseemTheDream/7a72e4ac59f528133822

**Comment by jfking** [comment]
Oh for some reason I never saw that gist

**Comment by jfking** [comment]
These most recent changes are causing the angular app to fail to initialize. Is it the change to app.js?

File: parse/public/js/controllers/place.js [hunk 4]
```diff
@@ -40,9 +40,17 @@ angular.module('atlasApp').controller('PlaceCtrl', function(
    * Initalizes the Place controller.
    */
   $scope.init = function() {
-    $scope.resizeView();
-    $(window).resize($scope.resizeView);
-    $scope.initMap();
+
+    MapsService.initMap();
+    MapsService.resizeView();
```
Comment (WaseemTheDream) [inline parse/public/js/controllers/place.js]: Remove this line it's causing the crash I think.

File: parse/public/js/controllers/place.js [hunk 5]
```diff
@@ -61,58 +69,23 @@ angular.module('atlasApp').controller('PlaceCtrl', function(
       });
     }
 
-  }
+    $scope.locationProvider = new LocationProvider();
```
Comment (WaseemTheDream) [inline parse/public/js/controllers/place.js]: Doesn't LocationProvider take a Map object?

File: parse/public/js/MapsService.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    });
+
+    return position;
+  };
+
+
+  /**
+   * Resizes the view to fit within the bounds of the screen.
+   */
+  MapsService.setMapHeight = function(newHeight) {
```
Comment (WaseemTheDream) [inline parse/public/js/MapsService.js:50]: You need to expose it in the list of public functions.

**Comment by jfking** [comment]
Haha yeah such a dumb little thing

**Pull request merged**